core/src/main/java/org/apache/iceberg/deletes/BitmapPositionDeleteIndex.java

```diff
@@ -92,4 +107,113 @@ public Collection<DeleteFile> deleteFiles() {
+   * @param deleteFile the DV file
+   * @return the deserialized index
+   */
+  public static PositionDeleteIndex deserialize(byte[] bytes, DeleteFile deleteFile) {
+    ByteBuffer buffer = ByteBuffer.wrap(bytes);
+    int bitmapDataLength = readBitmapDataLength(buffer, deleteFile);
+    RoaringPositionBitmap bitmap = deserializeBitmap(bytes, bitmapDataLength, deleteFile);
+    int crc = computeChecksum(bytes, bitmapDataLength);
+    int crcOffset = LENGTH_SIZE_BYTES + bitmapDataLength;
+    int expectedCrc = buffer.getInt(crcOffset);
+    Preconditions.checkArgument(crc == expectedCrc, "Invalid CRC");
+    return new BitmapPositionDeleteIndex(bitmap, deleteFile);
+  }
+
+  // computes and validates the length of the bitmap data (magic bytes + bitmap)
+  private static int computeBitmapDataLength(RoaringPositionBitmap bitmap) {
+    long length = MAGIC_NUMBER_SIZE_BYTES + bitmap.serializedSizeInBytes();
+    long bufferSize = LENGTH_SIZE_BYTES + length + CRC_SIZE_BYTES;
+    Preconditions.checkState(bufferSize <= Integer.MAX_VALUE, "Can't serialize index > 2GB");
+    return (int) length;
+  }
+
+  // serializes the bitmap data (magic bytes + bitmap) using the little-endian byte order
+  private static void serializeBitmapData(
+      byte[] bytes, int bitmapDataLength, RoaringPositionBitmap bitmap) {
+    ByteBuffer bitmapData = pointToBitmapData(bytes, bitmapDataLength);
+    bitmapData.putInt(MAGIC_NUMBER);
+    bitmap.serialize(bitmapData);
+  }
+
+  // points to the bitmap data in the blob
+  private static ByteBuffer pointToBitmapData(byte[] bytes, int bitmapDataLength) {
+    ByteBuffer bitmapData = ByteBuffer.wrap(bytes, BITMAP_DATA_OFFSET, bitmapDataLength);
+    bitmapData.order(ByteOrder.LITTLE_ENDIAN);
+    return bitmapData;
+  }
+
+  // checks the blob size is equal to the bitmap data length + extra bytes for length and CRC
+  private static int readBitmapDataLength(ByteBuffer buffer, DeleteFile deleteFile) {
+    int length = buffer.getInt();
+    long expectedLength = deleteFile.contentSizeInBytes() - LENGTH_SIZE_BYTES - CRC_SIZE_BYTES;
+    Preconditions.checkArgument(
+        length == expectedLength,
+        "Invalid bitmap data length: %s, expected %s",
+        length,
+        expectedLength);
+    return length;
+  }
+
+  // validates magic bytes and deserializes the bitmap
+  private static RoaringPositionBitmap deserializeBitmap(
+      byte[] bytes, int bitmapDataLength, DeleteFile deleteFile) {
+    ByteBuffer bitmapData = pointToBitmapData(bytes, bitmapDataLength);
+    int magicNumber = bitmapData.getInt();
+    Preconditions.checkArgument(
+        magicNumber == MAGIC_NUMBER,
+        "Invalid magic number: %s, expected %s",
+        magicNumber,
+        MAGIC_NUMBER);
+    RoaringPositionBitmap bitmap = RoaringPositionBitmap.deserialize(bitmapData);
+    long cardinality = bitmap.cardinality();
+    long expectedCardinality = deleteFile.recordCount();
```

Comment:
NIt: I feel like we could inline ` bitMap.cardinality() == deleteFile.recordCount()` in the check below and the intent of the check would still be clear

Reply:
Did this for consistency with the method above. It shouldn't be expensive to call `recordCount()` twice, so I am OK changing it.

Reply:
Let me actually keep it to avoid one more round of CI.